server.py

```diff
@@ -234,33 +234,8 @@ def process_course(raw_course):
         raise ScrapeError(
             "malformed course code: {}".format(repr(course_code)))
     department, course_number, num_suffix, school, section = match.groups()
-    if not department:
-        raise ScrapeError("empty string for department")
-    if "/" in department:
-        raise ScrapeError("department contains slashes: {}"
-                          .format(repr(department)))
-    try:
-        course_number = int(course_number)
-    except ValueError:
-        raise ScrapeError(
-            "malformed course number: {}".format(repr(course_number)))
-    if course_number <= 0:
-        raise ScrapeError(
-            "non-positive course number: {}".format(course_number))
-    if "/" in num_suffix:
-        raise ScrapeError("course code suffix contains slashes: {}"
-                          .format(repr(num_suffix)))
-    if not school:
-        raise ScrapeError("empty string for school")
-    if "/" in school:
-        raise ScrapeError("school contains slashes: {}".format(repr(school)))
-    try:
-        section = int(section)
-    except ValueError:
```

Comment:
As for your question on why this is removed, the section variable corresponds to [0-9]+ regex group. It will be able to convert to int.

Reply:
You have changed the behavior of this code and broken the documented API guarantee of strictly positive section numbers. If that was your intention, then it needs to be documented and justified.

Reply:
Consider line 236 and 226: the section variable comes from regular expression matching using pattern `r"([A-Z]+) *?([0-9]+) *([A-Z]*[0-9]?) *([A-Z]{2})-([0-9]+)"` The pattern corresponding to the last group,`[0-9]+`, which cannot be negative. Because the regex has already guaranteed the section to be positive, we do not have to recheck.

Reply:
The API documents that section numbers are strictly positive. This means that they cannot be zero. Non-negative is not the same thing as strictly positive.

Reply:
I will modify that then. Sorry for the trouble.